src/main/java/games/strategy/engine/framework/map/download/DownloadFile.java

```diff
@@ -35,53 +37,63 @@ DownloadFileDescription getDownload() {
   }
 
   void startAsyncDownload() {
-    final File fileToDownloadTo = ClientFileSystemHelper.createTempFile();
-    fileToDownloadTo.deleteOnExit();
-    final FileSizeWatcher watcher = new FileSizeWatcher(
-        fileToDownloadTo,
-        bytesReceived -> downloadListener.downloadUpdated(download, bytesReceived));
     state = DownloadState.DOWNLOADING;
-    createDownloadThread(watcher).start();
+    createDownloadThread().start();
   }
 
   /*
    * Creates a thread that will download to a target temporary file, and once
    * complete and if the download state is not cancelled, it will then move
    * the completed download temp file to: 'downloadDescription.getInstallLocation()'
    */
-  private Thread createDownloadThread(final FileSizeWatcher watcher) {
+  private Thread createDownloadThread() {
     return new Thread(() -> {
-      final File fileToDownloadTo = watcher.getFile();
-      if (state != DownloadState.CANCELLED) {
-        final String url = download.getUrl();
-        try {
-          DownloadUtils.downloadToFile(url, fileToDownloadTo);
-        } catch (final Exception e) {
-          ClientLogger.logError("Failed to download: " + url, e);
-        }
-        if (state == DownloadState.CANCELLED) {
-          return;
-        }
-        state = DownloadState.DONE;
-        try {
-          Files.move(fileToDownloadTo, download.getInstallLocation());
-
-          final DownloadFileProperties props = new DownloadFileProperties();
-          props.setFrom(download);
-          DownloadFileProperties.saveForZip(download.getInstallLocation(), props);
-
-        } catch (final Exception e) {
-          final String msg = "Failed to move downloaded file (" + fileToDownloadTo.getAbsolutePath() + ") to: "
-              + download.getInstallLocation().getAbsolutePath();
-          ClientLogger.logError(msg, e);
-        }
+      if (state == DownloadState.CANCELLED) {
+        return;
+      }
+
+      final File tempFile = newTempFile();
+      final FileSizeWatcher watcher = new FileSizeWatcher(
+          tempFile,
+          bytesReceived -> downloadListener.downloadUpdated(download, bytesReceived));
+      try {
+        DownloadUtils.downloadToFile(download.getUrl(), tempFile);
+      } catch (final IOException e) {
+        ClientLogger.logError("Failed to download: " + download.getUrl(), e);
+        return;
+      } finally {
+        watcher.stop();
```

Comment:
This is the line that fixes the problem discussed in the PR description.  The key is to call `FileSizeWatcher#stop()` before the call to `Files#move()` below.

Everything else is just refactoring.